docs/flows.md

```diff
@@ -0,0 +1,90 @@
+# Common Authentication Data Flows
+
+## Authorization Creation (ie. Login)
+1. Client authenticates without JWT. (ie. email/password, oauth, etc)
+2. Invalidate any existing authorizations and refresh tokens for that user/client combo.
+3. Create a new authorization record in the database with
+  - id (unique)
+  - entityId (unique)
+  - clientId (unique)
+  - permissions (maybe if they get default permissions)
+   
+  _A authorization can have any number of verifying ids/values (ie. we could also add an organizationId, MFA pin, etc. into the mix)_
+
+4. Create an access token (JWT) with the verification values/ids:
+  - authorizationId
+  - entityId (verification values)
+  - clientId
+  
+  **NOTE:** This could now actually have a long TTL
+5. Create a refresh token with a very long TTL
+  - refreshId
+  - entityId (verification values)
+  - clientId
+6. Send back the access token (JWT) and refresh token (JWT) to the client
+
+## Access Token Verification
+The access token (JWT) is passed via an `Authorization` header **or** is attached to the connected socket. These are requests made to anything but the `/authentication` service.
+
+1. Client sends valid JWT access token or it's attached to the connected socket
+2. Server verifies JWT (in our special auth middleware) to ensure it is not:
+  - expired (then it removes or flags authorization as revoked/expired)
+  - malformed
+  - tampered with
+3. Server decodes the token payload and passes it down the auth middleware chain
+4. Server looks up authorization record by the `authorizationId` and passes it down the auth middleware chain (populateAuthorization)
```

Comment:
@Download maybe @ekryski or @daffl can clarify for me if I'm mistaken, but the auth middleware chain will be called like a normal hook.  This will allow you to pass params that will customize which middleware will run.  You have complete control over the middleware.  Here's a contrived example:

```
app.service('myservice').hooks({
  before: {
    all: [app.authenticate(flow: 'stateless'), otherHook()]
  }
});

app.service('my-super-secure-service').hooks({
  before: {
    all: [app.authenticate(flow: 'strict'), otherHook()]
  }
});
```